### PR TITLE
No streams

### LINKED ISSUE #361: Bug: Unable to create streams when no streams exist
When attempting to add a new stream I'm consistently getting the following 404 error regardless of which type of stream I try to add:

```
{
  "detail": "create stream failed: max() arg is an empty sequence"
}
```

From some quick looking through the code, it looks like it might be coming from an error when building the stream object from create_stream here: https://github.com/micro-nova/AmpliPi/blob/main/amplipi/ctrl.py#L663 but I have not set up a full debug environment to verify.

### PR DESCRIPTION
**Pull request by linknum23**
Fixes #361

### CODE REVIEW

File: amplipi/ctrl.py [hunk 1]
```diff
@@ -662,7 +665,7 @@ def delete_group(self, gid: int) -> ApiResponse:
       return ApiResponse.error('delete group failed: {} does not exist'.format(gid))
 
   def _new_stream_id(self):
-    stream: Optional[models.Stream] = max(self.status.streams, key=lambda stream: stream.id)
+    stream: Optional[models.Stream] = max(self.status.streams, key=lambda stream: stream.id, default=None)
```
Comment (linknum23) [inline amplipi/ctrl.py:668]: This is the actual fix!

File: amplipi/ctrl.py [hunk 2]
```diff
@@ -421,6 +421,9 @@ def set_source(self, sid: int, update: models.SourceUpdate, force_update: bool =
               other_src.input = ''
             stream.disconnect()
             stream.connect(idx)
+          elif src.input and 'stream=' in src.input:
```
Comment (linknum23) [inline amplipi/ctrl.py:424]: Secondary fix, We may want to try to actuate this code somewhere?

**Comment by linknum23** [comment]
Minor change to make sure that we can handle adding a stream when no other streams exist. The bugfix was a single line of code. 

Main changes are to the test infrastructure to handle testing for this issue.

File: amplipi/ctrl.py [hunk 3]
```diff
@@ -421,6 +421,9 @@ def set_source(self, sid: int, update: models.SourceUpdate, force_update: bool =
...
+            # TODO: should this stream id validation happen in the Source model?
+            raise Exception(f'StreamID specified by "{src.input}" not found')
```
Comment (linknum23) [inline amplipi/ctrl.py:426]: Secondary fix, We may want to try to actuate this code somewhere?

**Comment by linknum23** [comment]
Alright fixed one more bug. I think this is good to go now!

**Comment by linknum23** [comment]
Nevermind, looks like I unerthed another related bug

**Comment by Lohrer** [comment]
I'm not sure what the new bug is but I played around with no streams and it seems fixed!

**Comment by linknum23** [comment]
Dare I say this is actually fixed now?

**Review by Lohrer — APPROVED** [review]
Rebase on develop please!

**Pull request merged**